src/plugins/mpath.c

```diff
@@ -156,360 +184,295 @@ gboolean bd_mpath_is_tech_avail (BDMpathTech tech, guint64 mode, GError **error)
  * Tech category: %BD_MPATH_TECH_BASE-%BD_MPATH_TECH_MODE_MODIFY
  */
 gboolean bd_mpath_flush_mpaths (GError **error) {
-    const gchar *argv[3] = {"multipath", "-F", NULL};
-    gboolean success = FALSE;
-    gchar *output = NULL;
+    struct dmmp_mpath **mps = NULL;
+    struct dmmp_context *ctx = NULL;
+    uint32_t mp_count = 0;
+    uint32_t i = 0;
+    gboolean rc = FALSE;
+    const char *name = NULL;
+    int dmmp_rc = DMMP_OK;
+    char err_msg_buff[_ERR_MSG_BUFF_SIZE];
 
-    if (!check_deps (&avail_deps, DEPS_MPATH_MASK, deps, DEPS_LAST, &deps_check_lock, error))
-        return FALSE;
+    assert (error != NULL);
```

Comment:
Please don't use ``assert()`` in a library.

Reply:
We need some error to be set and ``FALSE`` be returned instead.

Reply:
Error cannot be set due to NULL pointer. Will just return `FALSE`.